3rdParty/json-schema-validation/include/validation/events_from_slice.hpp

```diff
@@ -79,14 +81,51 @@ void from_value(Consumer& consumer,
             for (const auto& element : VPackObjectIterator(slice)) {
                 VPackSlice key = element.key;
                 if (key.isString()) {
-                    auto key_ref = key.stringRef();
-                    consumer.key(std::string_view(key_ref.data(), key_ref.length()));
-                    Recurse(consumer, element.value, options, &slice, ignore_special);
+                    auto ref = key.stringRef();
+                    auto view = std::string_view(ref.data(), ref.length());
+                    if (skip_special(view, special)) {
+                        continue;
+                    }
+                    consumer.key(std::string_view(view));
+                    Recurse(consumer, element.value, options, &slice, special);
                     consumer.member();
-                } else if (!ignore_special) {
+                } else if (special != arangodb::validation::SpecialProperties::None) {
                     std::string translated_key;
                     using namespace arangodb::basics;
                     uint8_t which = static_cast<uint8_t>(key.getUInt()) + VelocyPackHelper::AttributeBase;
+
+                    switch (which) {
+                        case VelocyPackHelper::KeyAttribute: {
+                            if (!enum_contains_value(special, arangodb::validation::SpecialProperties::Key)) {
+                                continue;
+                            }
+                            break;
+                        }
+                        case VelocyPackHelper::IdAttribute: {
+                            if (!enum_contains_value(special, arangodb::validation::SpecialProperties::Id)) {
+                                continue;
+                            }
+                            break;
+                        }
+                        case VelocyPackHelper::RevAttribute: {
+                            if (!enum_contains_value(special, arangodb::validation::SpecialProperties::Rev)) {
+                                continue;
+                            }
+                            break;
+                        }
+                        case VelocyPackHelper::FromAttribute:
+                            if (!enum_contains_value(special, arangodb::validation::SpecialProperties::From)) {
+                                continue;
+                            }
+                            break;
```

Comment:
out of curiosity, why isn't the `FromAttribute` case enclosed in a scope, but all other cases are?